examples/js-sub.c

```diff
@@ -130,21 +171,48 @@ int main(int argc, char **argv)
 
     if (s == NATS_OK)
     {
-        if (pull)
+        if (pull && async)
+        {
+            jsOpts.PullSubscribeAsync.MaxMessages = (int) total;
+            jsOpts.PullSubscribeAsync.NoWait = true;
+            jsOpts.PullSubscribeAsync.Timeout = 3600 * 1000; // 1 hour
+            jsOpts.PullSubscribeAsync.FetchSize = 17;
```

Comment:
this is really low value. We use between 100 and 1k, depending on client and at what threshold it starts getting diminishing returns on throughput.

Reply:
These are not the defaults, just example values. I'll change the example to explicitly set all values to their respective defaults, that'd be more intention-revealing. The default fetch size is 128. The rest are all 0s. 2/5 `KeepAhead` should default to 0, given non-trivial implications of setting it.